kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -340,64 +470,85 @@ jq --arg AGENT_SIZE $AGENT_SIZE '.properties.agentPoolProfiles[0].vmSize=$AGENT_
+validate_and_restore_cluster_definition $AZURESTACK_CONFIGURATION_TEMP $AZURESTACK_CONFIGURATION || exit $ERR_API_MODEL
 
-log_level -i "Completed building API model file based on passed stamp information and other parameters."
+log_level -i "Done building cluster definition."
 
 #####################################################################################
-# Section to generate ARM template using AKS Engine, login using Azure CLI and deploy the template.
+# azure-cli cloud
 # https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack
-HYBRID_PROFILE=2018-03-01-hybrid
-log_level -i "Register to AzureStack cloud using below command."
-retrycmd_if_failure 5 10 az cloud register -n $ENVIRONMENT_NAME --endpoint-resource-manager $TENANT_ENDPOINT --suffix-storage-endpoint $SUFFIXES_STORAGE_ENDPOINT --suffix-keyvault-dns $SUFFIXES_KEYVAULT_DNS --endpoint-active-directory-resource-id $ENDPOINT_ACTIVE_DIRECTORY_RESOURCEID --endpoint-active-directory $ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT --endpoint-active-directory-graph-resource-id $ENDPOINT_GRAPH_ENDPOINT
-log_level -i "Set Azure stack environment."
-retrycmd_if_failure 5 10 az cloud set -n $ENVIRONMENT_NAME
-log_level -i "Update cloud profile with value: $HYBRID_PROFILE."
-retrycmd_if_failure 5 10 az cloud update --profile $HYBRID_PROFILE
-
-if [ $IDENTITY_SYSTEM == "ADFS" ] ; then
+
+# azure-cli needs the "adfs" suffix
+if [ $IDENTITY_SYSTEM == "ADFS" ]; then ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT=${ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT}adfs; fi;
+
+log_level -i "Registering to Azure Stack cloud."
+retrycmd_if_failure 5 10 60 az cloud register \
+-n $ENVIRONMENT_NAME \
+--endpoint-resource-manager $TENANT_ENDPOINT \
+--suffix-storage-endpoint $SUFFIXES_STORAGE_ENDPOINT \
+--suffix-keyvault-dns $SUFFIXES_KEYVAULT_DNS \
+--endpoint-active-directory-resource-id $ENDPOINT_ACTIVE_DIRECTORY_RESOURCEID \
+--endpoint-active-directory $ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT \
+--endpoint-active-directory-graph-resource-id $ENDPOINT_GRAPH_ENDPOINT || exit $ERR_AZS_CLOUD_REGISTER
+
+log_level -i "Setting Azure Stack environment."
+retrycmd_if_failure 5 10 60 az cloud set -n $ENVIRONMENT_NAME || exit $ERR_AZS_CLOUD_ENVIRONMENT
+
+log_level -i "Updating cloud profile with value: $HYBRID_PROFILE."
+retrycmd_if_failure 5 10 60 az cloud update --profile $HYBRID_PROFILE || exit $ERR_AZS_CLOUD_PROFILE
+
+#####################################################################################
+# azure-cli login
+
+if [ $IDENTITY_SYSTEM == "ADFS" ]; then
     log_level -i "Login to ADFS environment using Azure CLI."
-    retrycmd_if_failure 5 10 az login --service-principal -u $SPN_CLIENT_ID  -p $PWD/$CERTIFICATE_PEM_LOCATION --tenant $TENANT_ID --output none
+    retrycmd_if_failure 5 10 60 az login --service-principal -u $SPN_CLIENT_ID --tenant $TENANT_ID --output none -p $CERTIFICATE_PEM_LOCATION || exit $ERR_AZS_LOGIN_ADFS
```

Comment:
nit pick Can we move long statement to multi line command as this is the only statement going off.

Reply:
This is consistent with all the other one-liners in the file